src/renderer/qss/js/qssFocusManager.js

```diff
@@ -0,0 +1,343 @@
+/**
+ * Defines a `focusManager` component for the QSS
+ *
+ * A focus manager which enables navigation between the various focusable elements
+ * using (Shift +) Tab and all arrow keys.
+ * Copyright 2017 Raising the Floor - International
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ * The research leading to these results has received funding from the European Union's
+ * Seventh Framework Programme (FP7/2007-2013) under grant agreement no. 289016.
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+
+/* global fluid jQuery */
+
+"use strict";
+(function (fluid, jQuery) {
+    var gpii = fluid.registerNamespace("gpii");
+
+    /**
+     * A special type of focusManager which enables navigation between the focusable
+     * elements using Tab, Shift + Tab, Arrow Up, Arrow Down, Arrow Left and the
+     * Arrow Right key. The navigation focus basically moves in the visual direction.
+     */
+    fluid.defaults("gpii.qss.qssFocusManager", {
+        gradeNames: ["gpii.qss.horizontalFocusManager", "gpii.qss.verticalFocusManager"],
+
+        maxElementsPerFocusGroup: 2,
+
+        styles: {
+            button: "fl-qss-button",
+            smallButton: "fl-qss-smallButton"
+        },
+
+        listeners: {
+            "onArrowUpPressed.impl": {
+                funcName: "gpii.qss.qssFocusManager.onArrowUpPressed",
+                args: ["{that}"]
+            },
+            "onArrowDownPressed.impl": {
+                funcName: "gpii.qss.qssFocusManager.onArrowDownPressed",
+                args: ["{that}"]
+            },
+            "onArrowLeftPressed.impl": {
+                funcName: "gpii.qss.qssFocusManager.onArrowLeftPressed",
+                args: ["{that}"]
+            },
+            "onArrowRightPressed.impl": {
+                funcName: "gpii.qss.qssFocusManager.onArrowRightPressed",
+                args: ["{that}"]
+            }
+        },
+
+        invokers: {
+            getFocusInfo: {
```

Comment:
@amb26, I suggest that we keep the current implementation for the Morphic 1.1 release. It has been tested thoroughly which has shown its stability. Also, the FocusManager has already been part of master for some time, so if we decide to merge PR 1.1 to master, this would not introduce the functionality, but rather an expansion to its current state. 
I’ve almost completed the desired refactoring but as this is a core functionality, I’m concerned that there might be some cases that I’m missing.
As there are higher priority items, I think it would be better if we address those first.
I’ll open a separate PR for the refactoring addressing all of the comments related to the `focusManager`, and in case I don’t have the time to finalise it, someone else, probably HyperAspect, can resume working on it at a later stage.

Reply:
No worries, @danailbd - I've written up this requirement under https://issues.gpii.net/browse/GPII-3685 so it can be referred to in future work. I agree we can go ahead merging this pull and table this for work at a future date.